types/src/transaction/analyzed_transaction.rs

```diff
@@ -0,0 +1,259 @@
+    fn eq(&self, other: &Self) -> bool {
+        self.hash == other.hash
+    }
+}
+
+impl Eq for AnalyzedTransaction {}
+
+impl Hash for AnalyzedTransaction {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        state.write(self.hash.as_ref());
+    }
+}
+
+impl From<Transaction> for AnalyzedTransaction {
+    fn from(txn: Transaction) -> Self {
+        match txn {
+            Transaction::UserTransaction(signed_txn) => match signed_txn.payload() {
+                TransactionPayload::EntryFunction(func) => {
+                    match (
+                        *func.module().address(),
+                        func.module().name().as_str(),
+                        func.function().as_str(),
+                    ) {
+                        (AccountAddress::ONE, "coin", "transfer") => {
+                            let sender_address = signed_txn.sender();
+                            let receiver_address = bcs::from_bytes(&func.args()[0]).unwrap();
+                            AnalyzedTransaction::analyzed_transaction_for_coin_transfer(
+                                signed_txn,
+                                sender_address,
+                                receiver_address,
+                                true,
+                            )
+                        },
+                        (AccountAddress::ONE, "aptos_account", "transfer") => {
+                            let sender_address = signed_txn.sender();
+                            let receiver_address = bcs::from_bytes(&func.args()[0]).unwrap();
+                            AnalyzedTransaction::analyzed_transaction_for_coin_transfer(
+                                signed_txn,
+                                sender_address,
+                                receiver_address,
+                                false,
+                            )
+                        },
+                        (AccountAddress::ONE, "aptos_account", "create_account") => {
+                            let sender_address = signed_txn.sender();
+                            let receiver_address = bcs::from_bytes(&func.args()[0]).unwrap();
+                            AnalyzedTransaction::analyzed_transaction_for_create_account(
+                                signed_txn,
+                                sender_address,
+                                receiver_address,
+                            )
+                        },
+                        _ => AnalyzedTransaction::new(
+                            Transaction::UserTransaction(signed_txn),
+                            vec![],
+                            vec![],
+                        ),
+                    }
+                },
+                _ => AnalyzedTransaction::new(
+                    Transaction::UserTransaction(signed_txn),
+                    vec![],
+                    vec![],
+                ),
+            },
+            _ => AnalyzedTransaction::new(txn, vec![], vec![]),
```

Comment:
`_ => todo!()` instead? we need to pass in the r/w set but we don't know how to calculate.